Environment_Engine/Compute/MergeBuildingElements.cs

```diff
@@ -57,5 +60,63 @@ public static Panel MergePanels(this Panel panel1, Panel panel2)
             
             return rtnPanel;
         }
+
+        [Description("Merges the geometry of two Environment Panels if they share same connected space together and get panel with bigger area, tries boolean geometry and returns panel with new geometry")]
+        [Input("panel1", "An Environment Panel to merge from")]
+        [Input("panel2", "A second Environment Panel to merge from")]
+        [Output("mergedPanel", "The Environment Panel with the  merged area between two panels, but with properties of larger panel")]
+        public static Panel MergePanels(this Panel panel1, Panel panel2, double tolerance = Tolerance.Distance)
+        {
+            Panel rtnPanel = null;
+
+            if (panel1.Area() > panel2.Area())
+                rtnPanel = panel1.Copy();
+            else
+                rtnPanel = panel2.Copy();
+
+            List<string> connectedSpaces = panel1.ConnectedSpaces;
+            connectedSpaces.AddRange(panel2.ConnectedSpaces);
+            connectedSpaces = connectedSpaces.Where(x => !x.Equals("-1")).ToList();
+            connectedSpaces = connectedSpaces.Distinct().ToList();
+
+            var result = connectedSpaces.All(panel2.ConnectedSpaces.Contains) && connectedSpaces.Count == panel2.ConnectedSpaces.Count;
+
+            if (result)
+            {
+                rtnPanel.ConnectedSpaces = new List<string>(connectedSpaces);
+
+                List<Panel> panels = new List<Panel>();
+                panels.Add(panel1);
+                panels.Add(panel2);
+
+                //List<Polyline> pLines = new List<Polyline>();
+                //pLines.Add(panel1.Polyline());
+                //pLines.Add(panel2.Polyline());
+
+                List<Polyline> pLines = panels.Select(x => x.Polyline()).ToList();
+                List<Polyline> newGeometry = pLines.BooleanUnion(tolerance);
+
+                if (newGeometry.Count < 1)
+                {
+                    Reflection.Compute.RecordError("Could not output geometry for panel");
+                    return null;
+                }
+
+                Polyline max = newGeometry[0];
+                foreach (Polyline pl in newGeometry)
+                {
+                    if (pl.Area() > max.Area())
+                        max = pl;
+                }
+
+                rtnPanel = rtnPanel.SetGeometry(max);
+
+                return rtnPanel;
+
+            }
+            Reflection.Compute.RecordError("Panels have different ConnectedSpaces will not merge");
```

Comment:
Can we change this error message to something a little more human friendly such as "These panels do not belong to the same space(s) and cannot be merged together"?

Reply:
done